In [6]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

In [17]:
# Import data
filename = 'data/train.csv'
raw_df = pd.read_csv(filename)

size = raw_df.shape[0]
print('Total size: ' + str(size))

raw_df.head()

Total size: 250000


Id Prediction  DER_mass_MMC  DER_mass_transverse_met_lep  DER_mass_vis  \
0  100000          s       138.470                       51.655        97.827   
1  100001          b       160.937                       68.768       103.235   
2  100002          b      -999.000                      162.172       125.953   
3  100003          b       143.905                       81.417        80.943   
4  100004          b       175.864                       16.915       134.805   

   DER_pt_h  DER_deltaeta_jet_jet  DER_mass_jet_jet  DER_prodeta_jet_jet  \
0    27.980                  0.91           124.711                2.666   
1    48.146               -999.00          -999.000             -999.000   
2    35.635               -999.00          -999.000             -999.000   
3     0.414               -999.00          -999.000             -999.000   
4    16.405               -999.00          -999.000             -999.000   

   DER_deltar_tau_lep  ...  PRI_met_phi  PRI_met_sumet  PRI_jet_num  \
0               3.064  ...       -0.277        258.733            2   
1               3.473  ...       -1.916        164.546            1   
2               3.148  ...       -2.186        260.414            1   
3               3.310  ...        0.060         86.062            0   
4               3.891  ...       -0.871         53.131            0   

   PRI_jet_leading_pt  PRI_jet_leading_eta  PRI_jet_leading_phi  \
0              67.435                2.150                0.444   
1              46.226                0.725                1.158   
2              44.251                2.053               -2.028   
3            -999.000             -999.000             -999.000   
4            -999.000             -999.000             -999.000   

   PRI_jet_subleading_pt  PRI_jet_subleading_eta  PRI_jet_subleading_phi  \
0                 46.062                    1.24                  -2.475   
1               -999.000                 -999.00                -999.000   
2               -999.000                 -999.00                -999.000   
3               -999.000                 -999.00                -999.000   
4               -999.000                 -999.00                -999.000   

   PRI_jet_all_pt  
0         113.497  
1          46.226  
2          44.251  
3           0.000  
4           0.000  

[5 rows x 32 columns]

In [14]:
def compute_missing_values(df, column_name):
    
    column = df[column_name].to_numpy()

    return column[np.where(column == -999.0)].shape[0]

In [16]:
for column in raw_df.columns[2:]:
    n_nan = compute_missing_values(raw_df, column)
    rel_nan = n_nan / size
    
    print('{column} is missing {n_nan} values, with a percentage of {rel_nan}'
              .format(column=column, n_nan=n_nan, rel_nan=rel_nan)
         )

DER_mass_MMC is missing 38114 values, with a percentage of 0.152456
DER_mass_transverse_met_lep is missing 0 values, with a percentage of 0.0
DER_mass_vis is missing 0 values, with a percentage of 0.0
DER_pt_h is missing 0 values, with a percentage of 0.0
DER_deltaeta_jet_jet is missing 177457 values, with a percentage of 0.709828
DER_mass_jet_jet is missing 177457 values, with a percentage of 0.709828
DER_prodeta_jet_jet is missing 177457 values, with a percentage of 0.709828
DER_deltar_tau_lep is missing 0 values, with a percentage of 0.0
DER_pt_tot is missing 0 values, with a percentage of 0.0
DER_sum_pt is missing 0 values, with a percentage of 0.0
DER_pt_ratio_lep_tau is missing 0 values, with a percentage of 0.0
DER_met_phi_centrality is missing 0 values, with a percentage of 0.0
DER_lep_eta_centrality is missing 177457 values, with a percentage of 0.709828
PRI_tau_pt is missing 0 values, with a percentage of 0.0
PRI_tau_eta is missing 0 values, with a percentage of 0.0
PRI_tau_p

We can see that several features have high percentages of missing values (> 40%), then we start by training the data without those features. Based on the information we will understand, we will next decid how to treat the *DER_mass_MMC* feature, which is missing ~15% of the values.